dependencies/dependencies.go

```diff
@@ -12,40 +13,47 @@ const InterpreterDepsKey = "interpreter"
 type Interface interface {
 	HTTPClient() (*http.Client, error)
 	SecretService() (SecretService, error)
+	URLValidator() (url.Validator, error)
 }
 
 type DefaultDependencies struct {
-	httpclient    *http.Client
-	secretservice SecretService
+	Deps Deps
+}
+
+type Deps struct {
+	HTTPClient    *http.Client
+	SecretService SecretService
+	URLValidator  url.Validator
 }
 
 func (d DefaultDependencies) HTTPClient() (*http.Client, error) {
-	if d.httpclient != nil {
-		return d.httpclient, nil
+	if d.Deps.HTTPClient != nil {
+		return d.Deps.HTTPClient, nil
 	}
 	return nil, errors.New(codes.Unimplemented, "http client uninitialized in dependencies")
 }
 
 func (d DefaultDependencies) SecretService() (SecretService, error) {
-	if d.secretservice != nil {
-		return d.secretservice, nil
+	if d.Deps.SecretService != nil {
+		return d.Deps.SecretService, nil
 	}
 	return nil, errors.New(codes.Unimplemented, "secret service uninitialized in dependencies")
 }
-
-// The values defined below come from the default implementation of Transport.
-// It establishes network connections as needed and caches them for reuse by subsequent calls.
-// It uses HTTP proxies as directed by the $HTTP_PROXY and $NO_PROXY (or $http_proxy and $no_proxy) environment variables.
-func NewDefaultDependencies() Interface {
-	return &DefaultDependencies{
-		httpclient:    nil,
-		secretservice: nil,
+func (d DefaultDependencies) URLValidator() (url.Validator, error) {
+	if d.Deps.URLValidator != nil {
+		return d.Deps.URLValidator, nil
 	}
+	return nil, errors.New(codes.Unimplemented, "url validator uninitialized in dependencies")
 }
 
-func NewDependenciesInterface(hc *http.Client, ss SecretService) Interface {
-	return &DefaultDependencies{
-		httpclient:    hc,
-		secretservice: ss,
+// NewDefaultDependecies produces a set of dependencies.
+// Not all dependencies have valid defaults and will not be set.
+func NewDefaultDependencies() DefaultDependencies {
```

Comment:
I think the default dependencies should be all-nil so that we get errors by default when we use them.  or else we need to have a special type of dependencies struct that has this behavior.   

Then we could add a `NewBasicDependencies` or something to represent this case.  

The issue is that there are a handful of places such as package evaluation that need to initialize dependencies to avoid nil references, but we don't want those dependencies to be functional in any way (e.g. loading a package should not send HTTP requests)
Using NewDependenciesInterface